ask/commands.py

```diff
@@ -0,0 +1,182 @@
+import sublime
+import sublime_plugin
+import pieces_os_client as pos_client
+from pieces.settings import PiecesSettings
+import re
+from difflib import Differ 
+from .prompts import *
+import mdpopups
+import time
+LIST_ITEMS = [
+	("Fix a bug",BUGS_PROMPT),
+	("Find a possible bug", POSSIBLE_BUGS_PROMPT),
+	("Write cleaner code", CLEANER_CODE_PROMPT),
+	("Add a doc string to this function",DOC_STRING_PROMPT),
+	("Add comments to the code",ADD_COMMENTS_PROMPT)
+] # TODO: Add more suff 
+
+
+class DescriptionInputHandler(sublime_plugin.TextInputHandler):
+	def placeholder(self):
+		return "Enter a quick description of the bug itself or useful error message"
+	def validate(self, text):
+		return len(text) > 0
+
+class QuestionInputHandler(sublime_plugin.ListInputHandler):
+	def list_items(self):
+		return LIST_ITEMS
+	def next_input(self,args):
+		if args in description_needed_commands:
+			return DescriptionInputHandler()
+
+
+
+
+class PiecesAskQuestionCommand(sublime_plugin.TextCommand):
+
+	def run(self,edit, question, description=None):
+		sublime.set_timeout_async(lambda:self.run_async(edit,question,description),0)
+
+		self.view.set_status('Pieces Refactoring', 'Copilot is thinking...')
+		
+
+	def run_async(self,edit, question, description):
+		# Get the current selection
+		self.selection = self.view.sel()[0]
+		selected_text = self.view.substr(self.selection)
+		
+		if "." in self.view.file_name():
+			self.langauge = self.view.file_name().split(".")[-1]
```

Comment:
is this proper? with the at index of '-1'?

Reply:
or did you want the "length of your array - 1"

Reply:
python list indexes are quite the thing. I'm pretty sure '-1' index is the last element of the list